dotCMS/src/integration-test/java/com/dotcms/security/apps/AppsUtilTest.java

```diff
@@ -0,0 +1,165 @@
+    }
+
+    /**
+     * https://www.w3schools.com/charsets/ref_html_utf8.asp
+     *
+     * @param codePoints char code see utf char codes.
+     * @return the utf string representation.
+     */
+    private static String fromCharCode(final int... codePoints) {
+        return new String(codePoints, 0, codePoints.length);
+    }
+
+    static class UTFCharsRangeTestCase {
+
+        final int fromCode;
+        final int toCode;
+        final String description;
+
+        UTFCharsRangeTestCase(final int fromCode, final int toCode, final String description) {
+            this.fromCode = fromCode;
+            this.toCode = toCode;
+            this.description = description;
+        }
+    }
+
+    /**
+     * This tests the two methods used to serialize a secret converting it into a json stored as chars
+     * then and putting it back together as an object.
+     * Given Scenario: A Random Secret object is constructed with a random alphanumeric string then serialize to bytes and back.
+     * Expected Result: The resulting object must match the original one passed in.
+     * @throws DotDataException
+     */
+    @Test
+    public void Test_Secret_Json_Serialization_No_String_Middle_Man() throws DotDataException {
+        final AppSecrets secretsIn = new AppSecrets.Builder()
+                .withKey("TheKey")
+                .withHiddenSecret("hidden1", "I'm hidden")
+                .withSecret("non-hidden1", "I'm not hidden")
+                .withSecret("non-hidden5", RandomStringUtils.randomAlphanumeric(2337))
+                .withSecret("bool1", true)
+                .build();
+        final char[] toJsonAsChars = toJsonAsChars(secretsIn);
+        final AppSecrets secretsOut = readJson(toJsonAsChars);
+        assertEquals(secretsIn.getKey(), secretsOut.getKey());
+
+        assertEquals(secretsIn.getSecrets().size(), secretsOut.getSecrets().size());
+
+        final Set<Entry<String, Secret>> secretsInEntries = secretsIn.getSecrets().entrySet();
+        for (final Entry<String, Secret> entryIn : secretsInEntries) {
+            assertEquals(secretsIn.getKey(), secretsOut.getKey());
+            final Secret out = secretsOut.getSecrets().get(entryIn.getKey());
+            assertNotNull(out);
+            assertTrue(out.equals(entryIn.getValue()));//This does a deepEquals.
+        }
+    }
+
+    /**
+     * This basically test the encryption decryption round trip that a text could take.
+     * Given Scenario:
+     * Expected Result:
+     * @throws EncryptorException
+     */
+    @Test
+    public void Test_Encrypt_Decrypt_Text_No_Middle_String() throws EncryptorException {
+        final Key key = Encryptor.generateKey();
+        final String input = RandomStringUtils.randomAlphanumeric(1000);
+        final char[] chars = AppsUtil.encrypt(key, input.toCharArray());
+        final char[] decrypted = AppsUtil.decrypt(key, new String(chars));
+        Assert.assertEquals(input, new String(decrypted));
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Unnecessary use of fully qualified name 'Assert.assertEquals' due to existing static import 'org.junit.Assert.assertEquals'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5329810)